src/intervals/interval_operations/pointwise_boolean.jl

```diff
@@ -0,0 +1,206 @@
+        possible outcome as a `BooleanInterval`.
+        This is safe but very strict, always erroring when an interval is used
+        in a conditional statement.
+    - `:ternary` (default) : With this politics return `missing` when the
+        boolean operation does not return the same answer for all elements
+        of the involved interval(s).
+        This only causes error in conditional statements when hitting `missing`.
+        When it does not error it is safe.
+
+The current pointwise politic can be changed by overriding the function
+`IntervalArithmetic.pointwise_politic()`.
+
+Example
+=======
+
+          | (-1..3) > 2   | (-1..3) <= 2  | (-1..1) > 2 | (-1..1) < 2 |
+----------|---------------|---------------|-------------|-------------|
+:is_all   | false         | false         | false       | true        |
+:interval | [true, false] | [true, false] | [false]     | [true]      |
+:ternary  | missing       | missing       | false       | true        |
+
+"""
+struct PointwisePolitic{P} end
+
+const pointwise_bool_operations = (
+    :(==), :(!=), :<, :(<=), :>, :(>=)
+)
+
+const pointwise_bool_functions = (
+    :isinf, :isfinite, :isinteger, :iszero
+)
+
+## :ternary
+function ==(::PointwisePolitic{:ternary}, x::Interval, y::Interval)
+    isthin(x) && isthin(y) && x.lo == y.lo && return true
+    (x.hi < y.lo || x.lo > y.hi) && return false
+    return missing
+end
+
+function <(::PointwisePolitic{:ternary}, x::Interval, y::Interval)
+    strictprecedes(x, y) && return true
+    precedes(y, x) && return false
+    return missing
+end
+
+function <=(::PointwisePolitic{:ternary}, x::Interval, y::Interval)
+    precedes(x, y) && return true
+    strictprecedes(y, x) && return false
+    return missing
+end
+
+!=(::PointwisePolitic{:ternary}, x::Interval, y::Interval) = !(==(PointwisePolitic{:ternary}(), x, y))
+>(::PointwisePolitic{:ternary}, x::Interval, y::Interval) = !<=(PointwisePolitic{:ternary}(), x, y)
+>=(::PointwisePolitic{:ternary}, x::Interval, y::Interval) = !<(PointwisePolitic{:ternary}(), x, y)
+
+# Boolean functions
+# NOTE this interacts with flavors.
+function isinf(::PointwisePolitic{:ternary}, x::Interval)
+    if contains_infinity(x)
+        isthing(x) && return true
+        return missing
+    end
+
+    return false
+end
+
+isfinite(::PointwisePolitic{:ternary}, x::Interval) = !isinf(PointwisePolitic{:ternary}(), x)
+iszero(::PointwisePolitic{:ternary}, x::Interval) = ==(PointwisePolitic{:ternary}(), x, 0)
+
+function isinteger(::PointwisePolitic{:ternary}, x::Interval)
+    (x.lo == x.hi) && isinteger(x.lo) && return true
+    floor(x.hi) < ceil(x.lo) && return false
+    return missing
+end
+
+
+## :interval
+"""
+    BooleanInterval
+
+Type representing a set containing `true` and/or `false`.
+
+Test what it contains using `in`. For example `true in BooleanInterval(true, false)`.
+"""
+struct BooleanInterval
+    has_true::Bool
```

Comment:
could be called `lo` and `hi` in analogy to normal intervals.

Reply:
The design I choose here is a bit different. There are 4 possibilities>
- `[false, true]` => `has_true == true && has_false == true` => represented by the two bits `11`
- `[false,]` => `has_true == false && has_false == true` => represented by the two bits `01`
- `[true]` => `has_true == true && has_false == false` => represented by the two bits `10`
- `[]` => `has_true == false && has_false == false` => represented by the two bits `00` (not allowed)

Maybe the compact memory layout does not justify the slightly confusing structure.

Another way of doing it was to have a single field as follow (`missing` representing `[false, true]`)
```
struct BooleanInterval
    ternary::Union{missing, Bool}
end
```